manticore/manticore.py

```diff
@@ -226,22 +227,28 @@ def register_plugin(self, plugin):
         #Global enumeration of valid events
         assert isinstance(plugin, Plugin)
         assert plugin not in self.plugins, "Plugin instance already registered"
-        assert plugin.manticore is None, "Plugin instance already owned" 
+        assert plugin.manticore is None, "Plugin instance already owned"
+
         plugin.manticore = self
         self.plugins.add(plugin)
-        
-        for event_name in Plugin.event_names:
-            callback_name = event_name+'_callback'
-            if hasattr(plugin, callback_name):
-                self.subscribe(event_name, getattr(plugin, callback_name))
-            
+
+        #XXX(yan): keeping track of all_events is potentially too heavyweight just for warnings.
+        all_events = set()
+        for src in (Executor, State, Cpu, Manticore):
```

Comment:
This means we cant make plugin that hooks on events emitted from the other subclasses of Eventful ?